tests/durability/test_disk_failure_recovery.py

```diff
@@ -520,3 +521,460 @@ def test_sns_repair_fail_disk_diff_cvg_equal_to_k(self):
+                                                              workload_info=workload_info)
+
+        LOGGER.info("COMPLETED: Test SNS repair works fine on near full system with failed disks "
+                    "are less than K(parity units)")
+
+    # pylint: disable=too-many-statements
+    @pytest.mark.data_durability
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-36397")
+    def test_near_full_sns_repair_fail_disk_diff_cvg_equal_to_k(self):
+        """
+        Validate SNS repair works fine on near full system with failed disks
+        are equal to K(parity units)
+        """
+        LOGGER.info("STARTED: Validate SNS repair works fine on near full system with failed disks "
+                    "are equal to K(parity units)")
+        near_full_percent = 60
+        users = self.mgnt_ops.create_account_users(nusers=1)
+        self.s3_clean = users
+        workload_info = None
+        s3userinfo = list(users.values())[0]
+
+        LOGGER.info("Step 1: Perform Write operations till overall disk space is filled %s",
+                    near_full_percent)
+        resp = self.dsk_rec_obj.get_user_data_space_in_bytes(self.node_master_list[0],
+                                                             near_full_percent)
+        assert_utils.assert_true(resp[0], resp[1])
+
+        if not resp[1]:
+            LOGGER.info("Current Memory usage is already more than expected memory usage,"
+                        " skipping write operation")
+        else:
+            workload_info = self.dsk_rec_obj.perform_near_full_sys_writes(s3userinfo=s3userinfo,
+                                                                          user_data_writes=resp[1],
+                                                                          bucket_prefix=
+                                                                          self.test_prefix[-1])
+
+        LOGGER.info("Step 2: Do IOs(Write and Read)")
+        self.test_prefix.append('test-36397')
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=s3userinfo,
+                                                    log_prefix=self.test_prefix[-1],
+                                                    skipcleanup=True)
+        assert_utils.assert_true(resp[0], resp[1])
+
+        LOGGER.info("Step 3: Get degraded byte count before failing the disk")
+        degraded_byte_cnt_before = self.dsk_rec_obj.get_byte_count_hctl(self.hlth_master_list[0],
+                                                                        "degraded_byte_count")
+        LOGGER.info("degraded byte count: %s", degraded_byte_cnt_before)
+
+        LOGGER.info("Step 4: Fail disks equal to K(parity units)")
+        LOGGER.info("No of parity units (K): %s", self.parity_units)
+        disk_fail_cnt = self.parity_units
+
+        resp = self.dsk_rec_obj.fail_disk(disk_fail_cnt, self.node_master_list[0],
+                                          self.node_worker_list, self.pod_name, on_diff_cvg=True)
```

Comment:
If no. cvg < disk_fail_cnt, will this work?
I think in that case you will need to decrease disk_fail_cnt as Ravindra did in other tests.

Reply:
disk failure count is equal to parity_units. 
As per EC configuration, SNS (N+K+S) will always be less than or equal to no of nodes/CVG. 
Hence, no. cvg < disk_fail_cnt  is not a valid condition to be taken care of.